src/main/java/TiTi/util/Parser.java

```diff
@@ -0,0 +1,124 @@
+package TiTi.util;
+
+import TiTi.task.Task;
+import TiTi.task.Event;
+import TiTi.task.Deadline;
+import TiTi.task.ToDo;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Make sense of user command, and return Response
+ * to tell Ui how to interact with user.
+ */
+public class Parser {
+    private SavedHistory savedHistory;
+    private TaskList taskList;
+    private Scanner sc;
+
+    /**
+     * Constructor for Parser class.
+     *
+     * @param savedHistory loads and writes to the saved data
+     * @param taskList list of current tasks
+     */
+    public Parser(SavedHistory savedHistory, TaskList taskList) {
+        this.savedHistory = savedHistory;
+        this.taskList = taskList;
+        sc = new Scanner(System.in);
+    }
+
+    /**
+     * Ask and interpret user command, and update list as required.
+     *
+     * @return Response object to tell Ui how to respons to user command
+     */
+    public Response cue() {
+        String input = sc.nextLine();
+        String cue = input.split(" ", 2)[0];
+
+        // handle missing description
+        if ((cue.equals("todo") || cue.equals("deadline")
+                || cue.equals("event") || cue.equals("find"))
+                && input.split(" ", 2).length == 1) {
+            return new Response(Response.Cue.MISSINGDESCRIPTION);
+        }
+
+        int taskNumber;
+        int numberOfTasks = taskList.size();
+        String[] temp;
+        String description;
+        String time;
+
+        switch (cue) {
+        case "bye":
+            savedHistory.saveHistory(taskList);
+            return new Response(Response.Cue.EXIT);
+
+        case "list":
+            return new Response(Response.Cue.LIST);
+
+        case "done":
+            taskNumber = Integer.parseInt("" + input.charAt(5));
+            if (taskNumber > numberOfTasks) {
+                return new Response(Response.Cue.TASKERROR);
+            } else {
+                Task task = taskList.get(taskNumber - 1);
+                task.complete();
+                return new Response(Response.Cue.DONE, task);
+            }
```

Comment:
```suggestion
            if (taskNumber > numberOfTasks) {
                return new Response(Response.Cue.TASKERROR);
            }
            Task task = taskList.get(taskNumber - 1);
            task.complete();
            return new Response(Response.Cue.DONE, task);
            
```
Since a new `Response` is returned in the if-block, the else-block is not required and can be removed so the main path is un-indented.
I noticed the same issue in several other places too.